packages/office-ui-fabric-react/src/components/DetailsList/DetailsFooter.base.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import { BaseComponent, classNamesFunction } from '../../Utilities';
+import { IDetailsFooterProps, IDetailsFooterStyleProps, IDetailsFooterStyles } from './DetailsFooter.types';
+
+const getClassNames = classNamesFunction<IDetailsFooterStyleProps, IDetailsFooterStyles>();
+
+export class DetailsFooterBase extends BaseComponent<IDetailsFooterProps, {}> {
```

Comment:
There should not be a default implementation; it should *always* be your own custom component. Otherwise we wind up with a bad collection of random `props`.